Arm branch

Bennables opened the pull request:
New Subsystem "MagicMotion" to move arm motor. 
New Commands magicPos and magicPosReset
- magicPos rotates an arm a certain number of degrees
- magicPosReset takes the current position and sets it as 0
New Constants and file named "Gains" that relate to how "MagicMotion" works

Chvoong24 committed:
changed patjh

Chvoong24 committed:
auto test

Chvoong24 committed:
test circle path

Chvoong24 committed:
changed constraints to 0

Chvoong24 committed:
test

Chvoong24 committed:
made robot faster in teleop

Chvoong24 committed:
changed max velocity to 20, commented it out

Chvoong24 committed:
made vlelocity even faster for Anton

Chvoong24 committed:
changed slew rate

Chvoong24 committed:
added square path

Chvoong24 committed:
changed dimensions

Chvoong24 committed:
changed dimensions

Chvoong24 committed:
forward 1 m

Chvoong24 committed:
made a 1 m straigth path

Chvoong24 committed:
erge branch 'main' into autonomous

Chvoong24 committed:
Merge pull request #11 from gwhs/main

Merging main to autonomous

Chvoong24 committed:
changed dimensions

Chvoong24 committed:
changed dimensions

Chvoong24 committed:
changed dimmensions

Chvoong24 committed:
test

Chvoong24 committed:
test

Bennables committed:
added calliope

Bennables committed:
yesh

Chvoong24 committed:
merged main with auto

Chvoong24 committed:
PID controller stuff

Chvoong24 committed:
PID

Bennables committed:
arm

RobertTheBlair commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -44,6 +46,8 @@ public class RobotContainer {
   // Set IP to 10.57.12.11
   // Set RoboRio to 10.57.12.2
 
+  private final Arm armsub = new Arm(5);
```
Should use a named constant - `ARM_TALON_MOTOR_ID` or similar

RobertTheBlair commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -140,7 +144,7 @@ private void configureButtonBindings() {
         .onTrue(Commands.runOnce(poseEstimator::resetFieldPosition, drivetrainSubsystem));
 
-    controller.b().onTrue(autoAimLime.withTimeout(3));
+    controller.b().onTrue(new move(armsub, 1));
```
Should use upper case for class names

RobertTheBlair commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -173,13 +177,16 @@ private void configureButtonBindings() {
    *
    * @return the command to run in autonomous
    */
+  private int autoPath = 1;
```
This field not used?

RobertTheBlair commented on src/main/java/frc/robot/commands/arm/magicPos.java:
```diff
@@ -0,0 +1,61 @@
+import frc.robot.Constants;
+import frc.robot.subsystems.MagicMotion;
+ 
+public class magicPos extends CommandBase {
```
Needs to be upper case

RobertTheBlair commented on src/main/java/frc/robot/commands/arm/magicPos.java:
```diff
@@ -0,0 +1,61 @@
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
```
Should there be an end condition?

RobertTheBlair commented on src/main/java/frc/robot/commands/arm/magicPosReset.java:
```diff
@@ -0,0 +1,50 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.MagicMotion;
+ 
+public class magicPosReset extends CommandBase {
```
Upper case please

RobertTheBlair commented on src/main/java/frc/robot/commands/arm/magicPosReset.java:
```diff
@@ -0,0 +1,50 @@
+  // needed for isFinished
+  // private double returnSpeed;
+  // private double amps;
+  private double angle;
```
are the angle and isInterrupted fields going to be used?

RobertTheBlair commented on src/main/java/frc/robot/commands/autonomous/TestAutonomous.java:
```diff
@@ -19,10 +21,20 @@ public TestAutonomous(
 
     PPSwerveControllerCommand drive =
-        DrivetrainSubsystem.followTrajectory(drivetrain, poseEstimatorSystem, trajectories.get(0));
+        new PPSwerveControllerCommand(
```
This branch is doing two things - adding the arm and changing the drivetrain auto stuff.

RobertTheBlair commented on src/main/java/frc/robot/commands/move.java:
```diff
@@ -0,0 +1,53 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Arm;
+
+public class move extends CommandBase {
```
upper case.
Also needs a name like "MoveArm"

RobertTheBlair commented on src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -349,6 +350,10 @@ public Command createCommandForTrajectory(Trajectory trajectory, Supplier<Pose2d
     return swerveControllerCommand;
   }
 
+  public PIDController getThetaController() {
```
we may need to create a new PIDController each time.  If we see situations where the first use behaves correctly, but the second does not, this would be a likely cause.

RobertTheBlair reviewed (changes requested):
looks like it's a work in progress branch.

Make sure to rename classes, and complete any partially done items

Bennables committed:
Fixed Arm commands to be use Sequentially

Bennables committed:
Added MagicMotionAbsoluteZero Command and Bore Encoder Subsystem.

Bennables committed:
????

Commit:
[armBranch] Merge branch 'main' into armBranch

# Conflicts:
#	src/main/java/frc/robot/RobotContainer.java
#	src/main/java/frc/robot/commands/Lime/AutoAimLime.java
#	src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

Commit:
[armBranch] - file rename for safety

Commit:
[armBranch] - revert name, so it has correct capitalization

RobertTheBlair commented on src/main/java/frc/robot/subsystems/ArmSubsystems/MagicMotion.java:
```diff
@@ -0,0 +1,102 @@
+    resetPosition();
+  }
+
+  public void robotInit() {
```
This function is not called anywhere. Where in the sample code was it called?

Commit:
[armBranch] - remove unused class

RobertTheBlair commented:
I fixed up the branch - the problem was a bunch of autonomous mode code was mixed in with the arm code.
Please remove the special code added for auto, and the path planner files.

Bennables committed:
changed ame

Bennables committed:
saved

Bennables committed:
Merge branch 'main' into armBranch

Bennables committed:
Switched command folder name from arm to Arm

Bennables committed:
saved

Bennables committed:
removign unused autos

Bennables committed:
inverted and correct values, still needs tuning

Bennables committed:
increasing speed

Bennables committed:
Merge branch 'main' into armBranch

Bennables committed:
fixing merge

Bennables committed:
fixing

RobertTheBlair commented on src/main/java/frc/robot/subsystems/ArmSubsystems/BoreEncoder.java:
```diff
@@ -0,0 +1,62 @@
+import java.util.Map;
+
+public class BoreEncoder extends SubsystemBase {
+  private final Encoder m_encoder = new Encoder(7, 8, false, CounterBase.EncodingType.k4X);
```
The IDs 7 and 8 should have named constants.

Bennables committed:
added constant for encoder ids

Bennables committed:
yes

RobertTheBlair reviewed (approved):
Good Job Michael
